examples/playwright/src/theia-notebook-cell.ts

```diff
@@ -0,0 +1,212 @@
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { ElementHandle, FrameLocator, Locator } from '@playwright/test';
+import { TheiaApp } from './theia-app';
+import { TheiaMonacoEditor } from './theia-monaco-editor';
+import { TheiaPageObject } from './theia-page-object';
+
+export type CellStatus = 'success' | 'error' | 'waiting';
+
+/**
+ * Page object for a Theia notebook cell.
+ */
+export class TheiaNotebookCell extends TheiaPageObject {
+
+    protected monacoEditor: TheiaEmbeddedMonacoEditor;
+
+    constructor(protected readonly locator: Locator, app: TheiaApp) {
+        super(app);
+        const editorLocator = locator.locator('div.theia-notebook-cell-editor');
+        this.monacoEditor = new TheiaEmbeddedMonacoEditor(editorLocator, app);
+    }
+
+    /**
+     * @returns The monaco editor page object of the cell.
+     */
+    get editor(): TheiaEmbeddedMonacoEditor {
+        return this.monacoEditor;
+    }
+
+    /**
+     * @returns `true` id the cell is a code cell, `false` otherwise.
+     */
+    async isCodeCell(): Promise<boolean> {
+        const classAttribute = await this.mode();
+        return classAttribute !== 'markdown';
+    }
+
+    /**
+     * @returns The mode of the cell, e.g. 'python', 'markdown', etc.
+     */
+    async mode(): Promise<string> {
+        this.locator.waitFor({ state: 'visible' });
+        const editorElement = await this.editor.locator.elementHandle();
+        if (editorElement === null) {
+            throw new Error('Could not find editor element for the notebook cell.');
+        }
+        const classAttribute = await editorElement.getAttribute('data-mode-id');
+        if (classAttribute === null) {
+            throw new Error('Could not find mode attribute for the notebook cell.');
+        }
+        return classAttribute;
+    }
+
+    /**
+     * @returns The text content of the cell editor.
+     */
+    async editorText(): Promise<string | undefined> {
+        const lines: string[] = [];
+        const linesCount = await this.monacoEditor.numberOfLines();
+        if (linesCount === undefined) {
+            return undefined;
+        }
+        for (let line = 1; line <= linesCount; line++) {
+            const lineText = await this.monacoEditor.textContentOfLineByLineNumber(line);
+            if (lineText === undefined) {
+                break;
+            }
+            lines.push(lineText);
+        }
```

Comment:
Would it make sense to move this into the `TheiaMonacoEditor`? I guess this functionality could be useful in general for the text editor and would then be easier to test and maintain, if it is at a single place.